examples/graphics/shader-graph-chunks-overrides.html

```diff
@@ -0,0 +1,215 @@
+        var app = new pc.Application(canvas);
+
+        // Set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+        app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+        app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+        window.addEventListener("resize", function () {
+            app.resizeCanvas(canvas.width, canvas.height);
+        });
+
+        var miniStats = new pcx.MiniStats(app);
+
+        // register core nodes
+        app.shaderNodes.register(pcsg.shaderNodes);
+
+        // set ambient light
+        app.scene.ambientLight = new pc.Color(0.2, 0.2, 0.2);
+
+        // Create an Entity with a camera component
+        camera = new pc.Entity();
+        camera.addComponent("camera", {
+            clearColor: new pc.Color(0.4, 0.45, 0.5)
+        });
+        camera.translate(0, 7, 25);
+        app.root.addChild(camera);
+
+        // Create an entity with a directional light component
+        var light = new pc.Entity();
+        light.addComponent("light", {
+            type: "directional",
+            color: new pc.Color(1, 1.0, 1.0)
+        });
+        app.root.addChild(light);
+        light.setLocalEulerAngles(15, -20, 0);
+
+        // load in a model
+        var url = "../assets/models/statue.glb";
+        app.assets.loadFromUrl(url, "container", function (err, asset) {
+            entityA = new pc.Entity();
+
+            var modelComponent = entityA.addComponent("model", {
+                    type: "asset",
+                    asset: asset.resource.model
+                });
+            var model = modelComponent.model;
+
+            app.root.addChild(entityA);
+           
+
+            entityB = entityA.clone();
+            app.root.addChild(entityB);
+            
+            entityC = entityA.clone();
+            app.root.addChild(entityC);
+
+            entityA.translate(-6, 0, 0);
+            entityB.translate( 0, 0, 0);
+            entityC.translate( 6, 0, 0);
+
+            // load a texture asset
+            app.assets.loadFromUrl("../assets/textures/logo.jpg", "texture", function (err, asset) {    
+                var logoMap = asset.resource;                
+                app.assets.loadFromUrl("../assets/textures/smiley.jpg", "texture", function (err, asset) {    
+                    var smileyMap = asset.resource;            
+                    app.assets.loadFromUrl("../assets/textures/particles-numbers.png", "texture", function (err, asset) {    
+                        var numbersMap = asset.resource;            
+
+                        // build one graph
+                        // create a shader graph builder
+                        var builder = new pc.ShaderGraphBuilder(app);
+
+                        // add input parameters
+                        var timeParam = builder.addParam('float', 'uTime', 0.0);
+                        var texParamA = builder.addParam('sampler2D', 'uTexMapA', logoMap);
```

Comment:
the value is described as optional (and should be optional), but crashes when not specified.